src/IO/WriteBufferFromVector.h

```diff
@@ -37,8 +39,8 @@ class WriteBufferFromVector : public WriteBuffer
     }
 
     /// Append to vector instead of rewrite.
-    WriteBufferFromVector(VectorType & vector_, AppendModeTag)
-        : WriteBuffer(nullptr, 0), vector(vector_)
+    WriteBufferFromVector(VectorType & vector_, AppendModeTag, UInt64 max_memory_usage_ = 0)
```

Comment:
Should it automatically check memory usage in `operator new[]`?

Reply:
Maybe it's because schema inference happens on storage creation outside the pipeline and we don't track memory on this stage (I don't know how a lot about our memory tracker)

Reply:
Memory tracking works everywhere, for each new/delete in clickhouse.
But what it does not track is malloc/free/mmap.
So if arrow uses one of above then the allocation will not be tracked.
I've looked briefly and tested correct Parquet format allocations tracked correctly, interesting.